src/gateway/Gateway.sol

```diff
@@ -134,36 +128,23 @@ contract Gateway is Auth {
     // --- Administration ---
     function file(bytes32 what, address data) public auth {
         if (what == "poolManager") poolManager = PoolManagerLike(data);
+        else if (what == "aggregator") aggregator = RouterAggregatorLike(data);
         else if (what == "investmentManager") investmentManager = InvestmentManagerLike(data);
         else revert("Gateway/file-unrecognized-param");
         emit File(what, data);
     }
 
-    function addIncomingRouter(address router) public auth {
-        incomingRouters[router] = true;
-        emit AddIncomingRouter(router);
-    }
-
-    function removeIncomingRouter(address router) public auth {
-        incomingRouters[router] = false;
-        emit RemoveIncomingRouter(router);
-    }
-
-    function updateOutgoingRouter(address router) public auth {
-        outgoingRouter = RouterLike(router);
-        emit UpdateOutgoingRouter(router);
-    }
-
     // --- Outgoing ---
     function send(bytes calldata message) public pauseable {
+        // TODO: check by message ID, that the origin matches
```

Comment:
This will be implemented in a followup PR, in the 2nd part of the message parsing refactor